packages/protobuf/src/to-plain-message.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable @typescript-eslint/no-explicit-any,@typescript-eslint/no-unsafe-assignment,@typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unsafe-return,@typescript-eslint/no-unsafe-argument,no-case-declarations */
+
+import { Message } from "./message.js";
+import type { AnyMessage, PlainMessage } from "./message.js";
+
+/**
+ * toPlainMessage returns a new object by striping
+ * all methods from a message, leaving only fields and
+ * oneof groups. It is recursive, meaning it applies this
+ * same logic to all nested message fields as well.
+ */
+export function toPlainMessage<T extends Message<T>>(
+  message: T
+): PlainMessage<T> {
+  if (typeof structuredClone === "function") {
+    return structuredClone(message) as PlainMessage<T>;
+  }
+  const type = message.getType();
+  const source = message as AnyMessage;
+  const target = {} as PlainMessage<AnyMessage>;
+  for (const member of type.fields.byMember()) {
+    const localName = member.localName;
+    switch (member.kind) {
+      case "scalar":
+      case "enum":
+        target[localName] = source[localName];
+        break;
+      case "message":
+        if (member.repeated) {
+          target[localName] = (source[localName] as any[]).map((val) =>
+            toPlainMessage(val)
+          );
+        } else if (source[localName] !== undefined) {
+          target[localName] = toPlainMessage(source[localName]);
```

Comment:
Can you add a test that uses [WrappersMessage](https://github.com/bufbuild/protobuf-es/blob/f955b2913d3186c31cbd129f8eb5f434af456c8b/packages/protobuf-test/extra/wkt-wrappers.proto#L20)?

It might be helpful too peek at [Message.clone](https://github.com/bufbuild/protobuf-es/blob/f955b2913d3186c31cbd129f8eb5f434af456c8b/packages/protobuf/src/private/util-common.ts#L181-L227).